hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java

```diff
@@ -288,6 +294,28 @@ public int getNumberOfContainers(PipelineID pipelineID) throws IOException {
     return stateManager.getNumberOfContainers(pipelineID);
   }
 
+  @Override
+  public void processReportedLeaderInfo(PipelineID pipelineID,
+      DatanodeDetails dn, ByteString leaderID) throws IOException {
+    Pipeline pipeline = stateManager.getPipeline(pipelineID);
+    Map<UUID, ByteString> ids =
+        reportedLeadersForPipeline.computeIfAbsent(pipelineID,
+            k -> new HashMap<>());
+
+    ids.put(dn.getUuid(), leaderID);
```

Comment:
It will be better to maintain node->leaderId map inside the Pipeline.java class. The information is pipeline specific and hence should be kept there.